go/api/base_client.go

```diff
@@ -3492,3 +3492,215 @@ func (client *baseClient) XClaimJustIdWithOptions(
+//	        Use `options.NewInfiniteStreamBoundary()` to specify an infinite stream boundary.
+//	opts  - Stream range options.
+//
+// Return value:
+//
+//	A `map` of key to stream entry data, where entry data is an array of
+//	pairings with format `[[field, entry], [field, entry], ...]`. Returns `nil` if `count` is non-positive.
+//
+// Example:
+//
+//	// Retrieve all stream entries
+//	res, err := client.XRangeWithCount(
+//		"key",
+//		options.NewInfiniteStreamBoundary(options.NegativeInfinity),
+//		options.NewInfiniteStreamBoundary(options.PositiveInfinity),
+//		options.NewStreamRangeOptions().SetCount(10),
+//	)
+//	fmt.Println(res) // Output: map[key:[["field1", "entry1"], ["field2", "entry2"]]]
+//
+//	// Retrieve exactly one stream entry by id
+//	res, err := client.XRangeWithCount(
+//		"key",
+//		options.NewStreamBoundary(streamId, true),
+//		options.NewStreamBoundary(streamId, true),
+//		options.NewStreamRangeOptions().SetCount(1),
+//	)
+//	fmt.Println(res) // Output: map[key:[["field1", "entry1"]]
+//
+// [valkey.io]: https://valkey.io/commands/xrange/
+func (client *baseClient) XRangeWithOptions(
+	key string,
+	start options.StreamBoundary,
+	end options.StreamBoundary,
+	opts *options.StreamRangeOptions,
+) (map[string][][]string, error) {
+	args := []string{key, string(start), string(end)}
+	if opts != nil {
+		optionArgs, err := opts.ToArgs()
+		if err != nil {
+			return nil, err
+		}
+		args = append(args, optionArgs...)
+	}
+	result, err := client.executeCommand(C.XRange, args)
+	if err != nil {
+		return nil, err
+	}
+	return handleMapOfArrayOfStringArrayOrNilResponse(result)
+}
+
+// Returns stream entries matching a given range of IDs in reverse order.
+// Equivalent to `XRange` but returns entries in reverse order.
+//
+// See [valkey.io] for details.
+//
+// Parameters:
+//
+//	key   - The key of the stream.
+//	start - The start position.
+//	        Use `options.NewStreamBoundary()` to specify a stream entry ID and its inclusive/exclusive status.
+//	        Use `options.NewInfiniteStreamBoundary()` to specify an infinite stream boundary.
+//	end   - The end position.
+//	        Use `options.NewStreamBoundary()` to specify a stream entry ID and its inclusive/exclusive status.
+//	        Use `options.NewInfiniteStreamBoundary()` to specify an infinite stream boundary.
+//
+// Return value:
+//
+//	A `map` of key to stream entry data, where entry data is an array of
+//	pairings with format `[[field, entry], [field, entry], ...]`.
+//
+// Example:
+//
+//	// Retrieve all stream entries
+//	res, err := client.XRevRange(
+//		"key",
+//		options.NewInfiniteStreamBoundary(options.PositiveInfinity),
+//		options.NewInfiniteStreamBoundary(options.NegativeInfinity),
+//	)
+//	fmt.Println(res) // Output: map[key:[["field2", "entry2"], ["field1", "entry1"]]]
+//
+// [valkey.io]: https://valkey.io/commands/xrevrange/
+func (client *baseClient) XRevRange(
```

Comment:
This is an overload of `XRevRangeWithOptions`. Call that func with a `nil` value for `opt` rather than handling logic here.